theano/gpuarray/opt.py

```diff
@@ -358,9 +357,10 @@ def apply(self, fgraph):
                             [mapping[i] for i in node.inputs],
                             node.outputs)
                     finally:
-                        time_opts[lopt] += time.time() - t_opt
-                        self.new_opts.append(lopt)
+                        t_opt2 = time.time()
                     if new_ops:
+                        self.new_opts.append(lopt)
+                        time_opts[lopt] = max(time_opts[lopt], (t_opt2 - t_opt))
```

Comment:
Here the there might be opts that are being used repeatedly. In other words, lopts can be same opt in consecutive loops(of the `toposort` loop) . In that case, the time it takes might be different because the node is different. So, if i keep this line as 
`time_opts[lopt] =  t_opt2 - t_opt`
I get the profiling result that all the local optimizers put together takes considerable time less than the new GraphToGPU optimizer. If i take maximum (like the code above), the local optimizer exceeds time taken by GraphToGPU by a large margin. What is the better way to estimate this ?